cluster-autoscaler/provisioningrequest/checkcapacity/processor.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checkcapacity
+
+import (
+	"time"
+
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/apis/autoscaling.x-k8s.io/v1beta1"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqwrapper"
+)
+
+const (
+	defaultReservationTime = 10 * time.Minute
+	defaultExpirationTime  = 7 * 24 * time.Hour // 7 days
+	// defaultMaxUpdated is a limit for ProvisioningRequest to update conditions in one ClusterAutoscaler loop.
+	defaultMaxUpdated = 20
+)
+
+type checkCapacityProcessor struct {
+	now        func() time.Time
+	maxUpdated int
+}
+
+// NewCheckCapacityProcessor return ProvisioningRequestProcessor for Check-capacity ProvisioningClass.
+func NewCheckCapacityProcessor() *checkCapacityProcessor {
+	return &checkCapacityProcessor{now: time.Now, maxUpdated: defaultMaxUpdated}
+}
+
+// Process iterates over ProvisioningRequests and apply:
+// -BookingExpired condition for Provisioned ProvisioningRequest if capacity reservation time is expired.
+// -Failed condition for ProvisioningRequest that were not provisioned during defaultExpirationTime.
+// TODO(yaroslava): fetch reservation and expiration time from ProvisioningRequest
+func (p *checkCapacityProcessor) Process(provReqs []*provreqwrapper.ProvisioningRequest) error {
+	expiredProvReq := []*provreqwrapper.ProvisioningRequest{}
+	failedProvReq := []*provreqwrapper.ProvisioningRequest{}
+	for _, provReq := range provReqs {
```

Comment:
one think I just remembered, we should only do this for check capacity provreqs.

Reply:
Done